cpp/include/raft/core/device_resources.hpp

```diff
@@ -123,6 +124,11 @@ class device_resources : public resources {
 
   rmm::exec_policy_nosync& get_thrust_policy() const { return resource::get_thrust_policy(*this); }
 
+  const raft::comms::nccl_clique& get_nccl_clique_handle() const
```

Comment:
I don't think we need to call this a `nccl_clique_handle`. I'd prefer to just call it `nccl_clique` for consistency with the actual naming elsewhere.